cas-server-core/src/main/java/org/jasig/cas/CentralAuthenticationServiceImpl.java

```diff
@@ -283,6 +284,16 @@ public ServiceTicket grantServiceTicket(
         // This throws if no suitable policy is found
         getAuthenticationSatisfiedByPolicy(ticketGrantingTicket, new ServiceContext(service, registeredService));
 
+        final List<Authentication> authentications = ticketGrantingTicket.getChainedAuthentications();
+        final Principal principal = authentications.get(authentications.size() - 1).getPrincipal();
+
+        final Map<String, Object> principalAttrs = registeredService.getAttributeReleasePolicy().getAttributes(principal);
+        if (!registeredService.getAuthorizationStrategy().isServiceAccessAuthorizedForPrincipal(principalAttrs)) {

```

Comment:
why yes, the name suggests that we are doing something with the principal where in fact we are passing attributes. Fair point. I can certainly amend this to be more like `doPrincipalAttributesAuthorizeServiceAccess()` or something better if you have one in mind.